crates/ra_parser/src/grammar/expressions.rs

```diff
@@ -535,11 +535,22 @@ fn cast_expr(p: &mut Parser, lhs: CompletedMarker) -> CompletedMarker {
     m.complete(p, CAST_EXPR)
 }
 
+// test arg_list
+// fn assert_float(s: &str, n: f64) {}
+// fn foo() {
+//     assert_float(
+//         "1.797693134862315708e+308L",
+//         #[allow(clippy::excessive_precision)]
+//         #[allow(dead_code)]
+//         1.797_693_134_862_315_730_8e+308,
+//     );
+// }
```

Comment:
Inline tests should be minimal. Their purpose is to elucidate how the parsing code works exactly, and not exhaustively tests it.

So, I suggest adding

```
test arg_with_attr
fn main() {
    foo(#[attr] 92)
}
```

right above the `attributes::outer_attributes(p);` line

Reply:
Ok, I can make that change. Is there another place for exhaustive tests?

Reply:
I think that this test should be enough, but you can drop additional tests to `test_data/parser/ok`. 

Let's also move the test to the line `547`, so it's close to the code it exercises.